### PR TITLE
[IMAGERES] Add Image Resource (imageres.dll)

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose
Maybe this DLL file is just an icon collection... I add downloads folder icon.
JIRA issue: [CORE-16118](https://jira.reactos.org/browse/CORE-16118)

icons

### CODE REVIEW

File: dll/win32/imageres/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+add_rc_deps(imageres.rc ${imageres_rc_deps})
+
+list(APPEND SOURCE imageres.c)
+
+add_library(imageres MODULE
+    ${SOURCE}
+    imageres.rc
+    ${CMAKE_CURRENT_BINARY_DIR}/imageres.def)
+
+set_module_type(imageres win32dll UNICODE)
```
Comment (learn-more) [inline dll/win32/imageres/CMakeLists.txt]: If you use `ENTRYPOINT 0` you can completely drop the `.c` file

Comment (learn-more) [reply dll/win32/imageres/CMakeLists.txt]: See for example https://git.reactos.org/?p=reactos.git;a=blob;f=dll/nls/normaliz/CMakeLists.txt;hb=ea8a6d6fb78629eab0d36291a24716623f3a05b8#l15

Comment (katahiromz) [reply dll/win32/imageres/CMakeLists.txt]: OK.

File: dll/win32/imageres/imageres_version.rc [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+ * PROJECT:     imageres
+ * LICENSE:     LGPL-2.1+ (https://spdx.org/licenses/LGPL-2.1+)
+ * PURPOSE:     Image Resource
+ * COPYRIGHT:   Copyright 2020 Katayama Hirofumi MZ (katayama.hirofumi.mz@gmail.com)
+ */
+
+#include "imageres_version.h"
+
+#define WINE_FILEVERSION WINE_FILEVERSION_MAJOR,WINE_FILEVERSION_MINOR,WINE_FILEVERSION_BUILD,WINE_FILEVERSION_PLATFORMID
+#define WINE_FILENAME_STR "imageres.dll"
```
Comment (HBelusca) [inline dll/win32/imageres/imageres_version.rc]: Why? Do you plan to send this dll to wine or to use their imageres?

Comment (katahiromz) [reply dll/win32/imageres/imageres_version.rc]: I'm Wine-friendly. If they want this file, I will provide it.

Comment (katahiromz) [reply dll/win32/imageres/imageres_version.rc]: I'm Wine-friendly. If they want this file, I will provide it.

**Comment by katahiromz** [comment]
At least, the Downloads folder uses this icon.

**Comment by learn-more** [comment]
From win10:
```
Searching for: imageres.dll
[HKCR\.library-ms\ShellNew]
  IconPath = %SystemRoot%\System32\imageres.dll,-1001
[HKCR\accountpicturefile\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-5203
[HKCR\Applications\explorer.exe]
  TaskbarGroupIcon = %SystemRoot%\System32\imageres.dll,-4
[HKCR\batfile\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-68

-- cut out a bunch of HKCR\CLSID\GUID stuff --

  IconReference = @%systemroot%\system32\imageres.dll,-1013
[HKCR\cmdfile\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-68
[HKCR\dbfile\DefaultIcon]
  (Default) = C:\Windows\system32\imageres.dll,-67
[HKCR\DeviceDisplayObject\InterfaceClass\{70FFD812-4C7F-4C7D-926A-637B7DD852AF}\Shell\System]
  Icon = imageres.dll,-149
[HKCR\dllfile\DefaultIcon]
  (Default) = C:\Windows\system32\imageres.dll,-67
[HKCR\giffile\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-71
[HKCR\inffile\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-69
[HKCR\inifile\DefaultIcon]
  (Default) = %SystemRoot%\system32\imageres.dll,-69
[HKCR\jpegfile\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-72
[HKCR\LibraryFolder\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-8
[HKCR\opensearchblocked\DefaultIcon]
  (Default) = %SystemRoot%\system32\imageres.dll,-105
[HKCR\opensearchdescription\DefaultIcon]
  (Default) = %SystemRoot%\system32\imageres.dll,-1025
[HKCR\Paint.Picture\DefaultIcon]
  (Default) = %systemroot%\system32\imageres.dll,-70
[HKCR\PhotoViewer.FileAssoc.Tiff\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-122
[HKCR\pjpegfile\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-72
[HKCR\pnffile\DefaultIcon]
  (Default) = %SystemRoot%\system32\imageres.dll,-67
[HKCR\pngfile\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-83
[HKCR\SearchConnectorFolder\DefaultIcon]
  (Default) = %SystemRoot%\system32\imageres.dll,-1025
[HKCR\SearchFolder\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-8
[HKCR\Stack\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-148
[HKCR\Stack.System.Music.AlbumID\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-1026
[HKCR\Stack.System.Music.AlbumTitle\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-1026
[HKCR\Stack.System.SearchScope\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-4
[HKCR\Stack.System.WordWheel\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-177
[HKCR\SystemFileAssociations\system\DefaultIcon]
  (Default) = %SystemRoot%\system32\imageres.dll,-67
[HKCR\SystemFileAssociations\text\DefaultIcon]
  (Default) = %SystemRoot%\system32\imageres.dll,-102
[HKCR\TIFImage.Document\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-122
[HKCR\txtfile\DefaultIcon]
  (Default) = %SystemRoot%\system32\imageres.dll,-102
[HKCR\Windows.IsoFile\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-5205
[HKCR\Windows.VhdFile\DefaultIcon]
  (Default) = %SystemRoot%\System32\imageres.dll,-32
```

**Comment by HBelusca** [comment]
image
Windows 2003 doesn't have it, so I'm not sure why we would need yet another resource-only library... :thinking:

**Comment by JoachimHenze** [comment]
I precise: Neither XPSP3 nor 2003SP2 do have this dll. I closed the PR, as I also see no gain in that.
I see no gain for adding desktop.ini s in general just to add those icons, especially not for those that do not exist in our target.

**Pull request closed without merging**